### PR TITLE
Add option to disable transition effects for the damage building module

### LINKED ISSUE #430: Damage Buildings: Add Option to Disable Destruction Effects
I suggest that for the Buildings -> Damage Buildings module you add an option under mean damage called Full (No Destruction Animation), and the just add to the function case 4: {_building setDamage [1, false]};  
This syntax was added in 1.67 and will destroy the building without the destruction effect, good for large scale destruction without crippling the server.

### PR DESCRIPTION
**Pull request by oOKexOo**
**When merged this pull request will:**
- Resolve #430

### CODE REVIEW

File: @AresModAchillesExpansion/addons/language_f/stringtable.xml [hunk 1]
```diff
@@ -2836,6 +2836,12 @@
                 <Chinesesimp>扩展</Chinesesimp>
                 <Chinese>擴展</Chinese>
             </Key>
+            <Key ID="STR_AMAE_DISABLE_TRANSITION_EFFECT">
+                <Original>Disable transition effect</Original>
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/language_f/stringtable.xml]: Disable destruction effect, would be more clearer.

File: @AresModAchillesExpansion/addons/modules_f_achilles/Buildings/functions/fn_BuildingsDestroy.sqf [hunk 2]
```diff
@@ -17,30 +15,45 @@ private _dialogResult =
...
 	case 1:
 	{
-		_buildings = nearestObjects [_center_pos, ["Building"], parseNumber (_dialogResult select 3), true];
+		_buildings = nearestObjects [_centerPos, ["Building"], _selectionRange, true];
 	};
 };
 
-private _mean_damage_type = _dialogResult select 1;
-private _distribution_type = _dialogResult select 2;
+params
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_achilles/Buildings/functions/fn_BuildingsDestroy.sqf]: ?

File: @AresModAchillesExpansion/addons/language_f/stringtable.xml (hunk 1 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/language_f/stringtable.xml]: ok

File: @AresModAchillesExpansion/addons/modules_f_achilles/Buildings/functions/fn_BuildingsDestroy.sqf (hunk 2 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/modules_f_achilles/Buildings/functions/fn_BuildingsDestroy.sqf]: Dunno what this is still doing here

**Comment by oOKexOo** [comment]
Oups, I must have added my new stringtable search tool to this PR by accident. I will remove it.

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf [hunk 3]
```diff
@@ -1,41 +1,52 @@
...
+		_meanMode		- <SCALAR> Mean damage type: 0 => none; 1 => slight; 2 => severe; 3 => full
+		_distMode		- <SCALAR> Type of distribution: 0 => delta; 1 => uniform; 2 => normal
+		_doSimulate		- <BOOL> If false, then destruction transition effects are disabled
+	
+	Returns:
+		None
+	
+	Examples:
+		(begin example)
+		[_buildings, 2, 2] call Achilles_fnc_damageBuildings;
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf]: Indent

File: @AresModAchillesExpansion/addons/modules_f_achilles/Buildings/functions/fn_BuildingsDestroy.sqf [hunk 4]
```diff
@@ -17,30 +15,40 @@ private _dialogResult =
 		[localize "STR_AMAE_SELECTION", [localize "STR_AMAE_NEAREST", localize "STR_AMAE_RANGE_NO_SI"]],
 		[localize "STR_AMAE_MEAN_DAMAGE",[localize "STR_AMAE_NO_DAMAGE",localize "STR_AMAE_LIGHT_DAMAGE",localize "STR_AMAE_SEVERE_DAMAGE",localize "STR_AMAE_FULL_DAMAGE"],2],
 		[localize "STR_AMAE_DISTRIBUTION",[localize "STR_AMAE_DELTA_DISTRIBUTION",localize "STR_AMAE_UNIFORM", localize "STR_AMAE_NORMAL_DISTRIBUTION"]],
-		[localize "STR_AMAE_RANGE","","100"]
+		[localize "STR_AMAE_RANGE","","100"],
+		[localize "STR_AMAE_DISABLE_DESTRUCTION_EFFECT",[localize "STR_AMAE_NO", localize "STR_AMAE_YES"]]
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_achilles/Buildings/functions/fn_BuildingsDestroy.sqf]: Yes then no.

File: @AresModAchillesExpansion/addons/modules_f_achilles/Buildings/functions/fn_BuildingsDestroy.sqf [hunk 5]
```diff
@@ -17,30 +15,40 @@ private _dialogResult =
 		[localize "STR_AMAE_SELECTION", [localize "STR_AMAE_NEAREST", localize "STR_AMAE_RANGE_NO_SI"]],
 		[localize "STR_AMAE_MEAN_DAMAGE",[localize "STR_AMAE_NO_DAMAGE",localize "STR_AMAE_LIGHT_DAMAGE",localize "STR_AMAE_SEVERE_DAMAGE",localize "STR_AMAE_FULL_DAMAGE"],2],
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_achilles/Buildings/functions/fn_BuildingsDestroy.sqf]: Add spacing between the commas (for all of them).
```suggestion
		[localize "STR_AMAE_MEAN_DAMAGE", [localize "STR_AMAE_NO_DAMAGE", localize "STR_AMAE_LIGHT_DAMAGE", localize "STR_AMAE_SEVERE_DAMAGE", localize "STR_AMAE_FULL_DAMAGE"], 2],
```

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf [hunk 6]
```diff
@@ -1,41 +1,52 @@
...
-params [["_buildings", [], [[]]], ["_mean_damage_type", 0, [0]], ["_distribution_type", 0, [0]]];
-
-private _fnc_getDamageType = switch (_distribution_type) do
+private _fnc_getDamageType = switch (_distMode) do
 {
-	case 0: {compile (str _mean_damage_type)};
+	case 0: {compile (str _meanMode)};
 	case 1: {{round (random 3)}};
-	case 2: {compile ("round (random [0," + (str _mean_damage_type) + ",3])")};
+	case 2: {compile ("round (random [0," + (str _meanMode) + ",3])")};
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf]: Add comments for what each type does.

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf [hunk 7]
```diff
@@ -1,41 +1,52 @@
...
+		Damages/Reinstates buildings
+	
+	Parameters:
+		_buildings		- <ARRAY> of <OBJECT> List of buildings to be processed
+		_meanMode		- <SCALAR> Mean damage type: 0 => none; 1 => slight; 2 => severe; 3 => full
+		_distMode		- <SCALAR> Type of distribution: 0 => delta; 1 => uniform; 2 => normal
+		_doSimulate		- <BOOL> If false, then destruction transition effects are disabled
+	
+	Returns:
+		None
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf]: Returns none or nothing?

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf (hunk 6 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf]: That's in the doc comment though

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf (hunk 7 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf]: I would suggest we use `none` like CBA does.

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf (hunk 3 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf]: Don't see anything off

File: @AresModAchillesExpansion/addons/modules_f_achilles/Buildings/functions/fn_BuildingsDestroy.sqf (hunk 4 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/modules_f_achilles/Buildings/functions/fn_BuildingsDestroy.sqf]: Ok, but default should still be `NO`

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf (hunk 3 above)
Comment (CreepPork) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf]: All your other examples had an indent, this one doesn't

File: @AresModAchillesExpansion/addons/modules_f_achilles/Buildings/functions/fn_BuildingsDestroy.sqf (hunk 4 above)
Comment (CreepPork) [reply @AresModAchillesExpansion/addons/modules_f_achilles/Buildings/functions/fn_BuildingsDestroy.sqf]: Correct.

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf (hunk 6 above)
Comment (CreepPork) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf]: So? You aren't going to put all the code comments into the doc block, are you? Complicated things should be explained. You aren't going to be the only one looking at this code.

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf (hunk 7 above)
Comment (CreepPork) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf]: Why?

Comment (neilzar) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf]: Nothing is the official Arma terminology, I prefer that over None.

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf [hunk 8]
```diff
@@ -1,41 +1,53 @@
...
-private _fnc_getDamageType = switch (_distribution_type) do
+private _fnc_getDamageType = switch (_distMode) do
 {
-	case 0: {compile (str _mean_damage_type)};
-	case 1: {{round (random 3)}};
-	case 2: {compile ("round (random [0," + (str _mean_damage_type) + ",3])")};
+	case 0: {compile (str _meanMode)};
+	case 1: {round (random 3)};
+	// Normally distributes mean damage for buildings
+	case 2: {compile ("round (random [0," + (str _meanMode) + ",3])")};
```
Comment (neilzar) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf]: I don't like that all the compiles are used here, they are a known cause for memory leaks.

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf [hunk 9]
```diff
@@ -49,22 +61,22 @@ private _fnc_getDamageType = switch (_distribution_type) do
 				if (count _other > 0) then
 				{
 					{
-						[_building, [_x,1]] remoteExecCall ["setHitIndex", 0];
+						[_building, [_x,1,_doSimulate]] remoteExecCall ["setHitIndex", 0];
```
Comment (neilzar) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf]:
```suggestion
						[_building, [_x, 1, _doSimulate]] remoteExecCall ["setHitIndex", 0];
```

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf [hunk 10]
```diff
@@ -49,22 +61,22 @@ private _fnc_getDamageType = switch (_distribution_type) do
...
 				if (_counter > 0) then
 				{
-					private _extend_count = ceil ((random 1) * _counter);
-					for "_i" from 1 to _extend_count do
+					private _extendCount = ceil ((random 1) * _counter);
+					for "_i" from 1 to _extendCount do
 					{
 						private _hitzone = selectRandom _hitzones;
-						[_building,  [_hitzone,1]] remoteExecCall ["setHitIndex", 0];
+						[_building,  [_hitzone,1,_doSimulate]] remoteExecCall ["setHitIndex", 0];
```
Comment (neilzar) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_damageBuildings.sqf]:
```suggestion
						[_building,  [_hitzone, 1, _doSimulate]] remoteExecCall ["setHitIndex", 0];
```

**Pull request merged**